change get categories through an endpoint

bomblito11 opened the pull request:
added redux operations to get categories
changed the way categories were fetched in addExpenseForm

bomblito11 committed:
change get categories through an endpoint

added redux operations to get categories
changed the way categories were fetched in addExpenseForm

KrzysztofOlejniczak commented on packages/frontend/src/redux/finance/operations.js:
```diff
@@ -82,3 +82,22 @@ export const deleteTransaction = createAsyncThunk(
+    const persistedToken = state.auth.token;
+    if (persistedToken === null) {
+      return thunkAPI.rejectWithValue('Token is missing');
+    }
```
get category enpoint not require auth token

KrzysztofOlejniczak reviewed (changes requested):
* get category enpoint not require auth token
* eslint warnings make build fail on netlify:
     - Line 6:3:   'fetchCategories' is defined but never used  no-unused-vars
     - Line 8:10:  'useEffect' is defined but never used        no-unused-vars
* you don't fetch categories: 
 
![image](https://github.com/KrzysztofOlejniczak/WalletApp/assets/109310116/4de3e722-ca6d-4e44-8a86-614fe9df0cb8)

bomblito11 committed:
Merge branch 'main' into feature/getCategories-frontend

bomblito11 committed:
bugfix

uncommented fetching
removed auth in get category endpoint
removed eslint warnings
